### PR TITLE
Component Tag in Transition Group, Fix #5095

### LINKED ISSUE #5095: Feature Request Allow Component Tag in Transition Group or Transition
Allow this kind of feature for transition-group or transition

```html
<transition-group tag="todo-layout-container">
    <!-- Which will render a component rather a tag -->
</transition-group>
```

and will compile to
```html

<todo-layout-container>

</todo-layout-container>
```

```js
let Todo = Vue.extend({
   name: 'todo-layout-container',
   methods: {
      //Per todo methods
   }
});
```

### PR DESCRIPTION
**Pull request by znck**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [x] It's submitted to the `dev` branch for v2.x (or to a previous version branch), _not_ the `master` branch
- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] All tests are passing: https://github.com/vuejs/vue/blob/dev/.github/CONTRIBUTING.md#development-setup
- [x] New/updated tests are included

If adding a **new feature**, the PR's description includes:
- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:** Close #5095

### CODE REVIEW

**Comment by posva** [comment]
ping @yyx990803

File: test/unit/features/transition/transition-group.spec.js [hunk 1]
```diff
@@ -293,5 +293,22 @@ if (!isIE9) {
       }).$mount()
       expect('<transition-group> children must be keyed: <div>').toHaveBeenWarned()
     })
+
+    it('component tag in transition group', () => {
+      const component = { template: '<div><slot></slot></div>' }
+      Vue.component('component', component)
```
Comment (yyx990803) [inline test/unit/features/transition/transition-group.spec.js]: Use a local component here so that it doesn't accidentally pollute other test cases.

Comment (znck) [inline test/unit/features/transition/transition-group.spec.js]: That is a problem. `<component>` is resolved in `<transition-group>`, if I register it in `App` it won't be available in `<transition-group>`.

> Created a feature request #5252

**Comment by Kingwl** [comment]
https://github.com/vuejs/vue/blob/f1c38674f28116e1e335873ecbc6716fb70448a3/src/core/vdom/create-element.js#L72

i have try resolveAsset parentComponent.$options here if context is transition-group
but this is not a good idea

**Comment by yyx990803** [comment]
If this only works for global components it's not really a complete fix. Currently the user can actually do

``` html
<transition-group :tag="$options.components.foo">
```

It renders correctly, but there will be patch errors on updates. I think there are thing that needs to be further debugged inside transition-group itself before we can tackle this request properly, but it's not really high priority at the moment.

**Pull request closed without merging**